core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -262,7 +262,9 @@ object RequestChannel extends Logging {
         m.responseSendTimeHist.update(Math.round(responseSendTimeMs))
         m.totalTimeHist.update(Math.round(totalTimeMs))
         m.requestBytesHist.update(sizeOfBodyInBytes)
-        m.messageConversionsTimeHist.foreach(_.update(Math.round(messageConversionsTimeMs)))
+        if (messageConversionsTimeMs > 0) {
```

Comment:
It will be better to check `messageConversionsTimeNanos` instead of the converted `ms`.